### PR TITLE
Making successful redirects more clear.

### PR DESCRIPTION
**Pull request by andrewconner**
Technically, `redirectToOrganization` and `redirectToOrganizationIntegrations` should just return paths, and the consumer should pick what it means (action performed? redirect?) @leogrim @ryanpbrewster

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/website/SlackController.scala [hunk 1]
```diff
@@ -318,7 +305,10 @@ class SlackController @Inject() (
 
 private sealed trait SlackResponse
 private object SlackResponse {
+  // Error, usually can't be elegantly recovered by clients.
   final case class Error(code: String) extends SlackResponse
+  // Client should blindly go to `url`, typically because they need to go to 3rd party to auth
   final case class RedirectClient(url: String) extends SlackResponse
-  case object ActionPerformed extends SlackResponse
+  // Requested action performed. Url included is a suggestion about a good place to go next, but clients can be smarter.
+  final case class ActionPerformed(url: Option[String] = None) extends SlackResponse
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/SlackController.scala:313]: Maybe explicitly call it `suggestedUrl: Option[String]`?

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/SlackController.scala:313]: Well, it's basically always wanted. Only in the case where the frontend can do better ("I know how to handle this kind of successful request"). I can rename it if you want

Comment (andrewconner) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/website/SlackController.scala:313]: `suggested` is implied by the type, though :smile:

**Comment by leogrim** [comment]
:tophat:

**Pull request merged**